waku/waku_relay/protocol.nim

```diff
@@ -210,8 +209,25 @@ proc subscribe*(w: WakuRelay, pubsubTopic: PubsubTopic, handler: WakuRelayHandle
     else:
       return handler(pubsubTopic, decMsg.get())
 
+  # rejects messages that are not WakuMessage
+  let wrappedValidator = proc(pubsubTopic: string, 
+                              message: messages.Message): Future[ValidationResult] {.async.} =
+    # can be optimized by checking if the message is a WakuMessage without allocating memory
+    # see nim-libp2p protobuf library
+    let msgRes = WakuMessage.decode(message.data)
+    if msgRes.isErr():
+      return ValidationResult.Reject
+    let msg = msgRes.get()
+
+    # now sequentially validate the message
+    if w.wakuValidators.hasKey(pubsubTopic):
+      let validatorsForTopic = w.wakuValidators[pubsubTopic]
+      for validator in validatorsForTopic:
+        return await validator(pubsubTopic, msg)
```

Comment:
not sure i understnad this return. if the first validator returns Accept, this `wrappedValidator` will return `Accept` and that it? no matter if the second or third validators reject the message? or am i missing something?

Reply:
i would expect something like 

```
for validator in validatorsForTopic:
  if (await validator(pubsubTopic, msg)) == ValidationResult.Reject:
    return ValidationResult.Reject
```

Reply:
Addressed in 449e96fd, thanks